src/mpcCoreKit.ts

```diff
@@ -427,13 +428,15 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
       storeWebBrowserFactor(deviceFactorKey, this);
       await this.inputFactorKey(new BN(deviceFactorKey, "hex"));
 
-      const backupFactorKey = await this.createFactor({ shareType: TssShareType.RECOVERY, ...enableMFAParams });
-
       const hashedFactorPub = getPubKeyPoint(hashedFactorKey);
       await this.deleteFactor(hashedFactorPub, hashedFactorKey);
       await this.deleteShareWithFactorKey(hashedFactorKey);
 
-      return backupFactorKey;
+      // only recovery factor = true
+      if (recoveryFactor) {
+        const backupFactorKey = await this.createFactor({ shareType: TssShareType.RECOVERY, ...enableMFAParams });
+        return backupFactorKey;
```

Comment:
If someone passes recovery factor to be `false` and clear browser storage, they are at a risk of losing their account right?

Reply:
Yes, this is requested by safe to be not to opinionated

Reply:
ok